### PR TITLE
Tidy up README and docs

### PR DESCRIPTION
**Pull request by glennmoy**

### CODE REVIEW

File: docs/src/testutils.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # TestUtils
 
-Provides test fakes, [`FakeTemplate`](@ref) and [`FakeModel`](@ref), that are useful for
-testing downstream dependencies, and [`test_interface`](@ref) for testing the Model's API
+Provides test fakes, `FakeTemplate` and `FakeModel`, that are useful for
+testing downstream dependencies, and `test_interface` for testing the model's API
```
Comment (nickrobinson251) [inline docs/src/testutils.md]: where'd the `@ref`s go?

File: docs/src/index.md [hunk 2]
```diff
@@ -2,14 +2,12 @@
...
-* Calling [`fit`](@ref) on a [`Template`](@ref).
-* Calling [`predict`](@ref) on a [`Model`](@ref).
-* Assigning traits such as [`EstimateTrait`](@ref) and [`OutputTrait`](@ref).
-* Testing interfaces and downstream dependencies with [`TestUtils`](@ref).
-
-For common examples of the interface being implemented see [BaselineModels.jl](https://gitlab.invenia.ca/invenia/research/BaselineModels.jl).
+* Calling `fit` on a `Template`.
+* Calling `predict` on a `Model`.
+* Assigning traits such as `EstimateTrait` and `OutputTrait`.
+* Testing interfaces and downstream dependencies with `TestUtils`.
```
Comment (nickrobinson251) [inline docs/src/index.md]: where are the `@ref`s???

File: docs/src/design.md [hunk 3]
```diff
@@ -46,30 +44,30 @@ outputs = StatsBase.predict(
...
    - `DistributionEstimate`: Estimates distributions over possible values.
- - `output_type` - determines how many output variates a model can learn
+ - `output_type` - determines how many output variates a `Model` can learn
    - `SingleOutput`: Fits and predicts on a single output only.
    - `MultiOutput`: Fits and predicts on multiple outputs at a time.
 
-The traits always agree between the model and the template.
-Every model and template should define all the listed traits.
+The traits always agree between the `Model` and the `Template`.
+Every `Model` and `Template` should define all the listed traits.
```
Comment (nickrobinson251) [inline docs/src/design.md]: all need `@ref`s

File: docs/src/design.md [hunk 4]
```diff
@@ -1,8 +1,6 @@
 ## Design Documentation
 
-This page details the key features of the design of BaselineModels.
-
-BaselineModels exists to solve the issue highlighted by following quote:
+This page details the key features of the design of `Models.jl`, which exists to solve the issue highlighted by following quote:
```
Comment (nickrobinson251) [inline docs/src/design.md]:
```suggestion
This page details the key features of the design of [Models.jl](https://github.com/invenia/Models.jl), which exists to solve the issue highlighted by following quote:
```

File: docs/src/design.md [hunk 5]
```diff
@@ -13,22 +11,22 @@ BaselineModels exists to solve the issue highlighted by following quote:
 
 -- [Sculley et al 2015](https://papers.nips.cc/paper/5656-hidden-technical-debt-in-machine-learning-systems)
 
-BaselineModels provides a common API for mostly preexisting models to allow them to all be used in the same way.
+`Models.jl` provides a common API for mostly preexisting models to allow them to all be used in the same way.
```
Comment (nickrobinson251) [inline docs/src/design.md]:
```suggestion
Models.jl provides a common API for mostly preexisting models to allow them to all be used in the same way.
```

File: README.md [hunk 6]
```diff
@@ -1,15 +1,16 @@
 # Models
 
-[![Latest](https://img.shields.io/badge/docs-latest-blue.svg)](https://invenia.pages.invenia.ca/research/Models.jl/)
-[![Coverage](https://gitlab.invenia.ca/invenia/Models.jl/badges/master/coverage.svg)](https://gitlab.invenia.ca/invenia/Models.jl/commits/master)
+[![Build Status](https://travis-ci.com/invenia/Models.jl.svg?branch=master)](https://travis-ci.com/invenia/Models.jl)
+[![Build Status](https://ci.appveyor.com/api/projects/status/github/invenia/Models.jl?svg=true)](https://ci.appveyor.com/project/invenia/Models-jl)
```
Comment (nickrobinson251) [inline README.md]: we don't need appveyor CI, so we don't need the badge either

File: README.md [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+[![Codecov](https://codecov.io/gh/invenia/Models.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/invenia/Models.jl)
+[![PkgEval](https://juliaci.github.io/NanosoldierReports/pkgeval_badges/N/Models.svg)](https://juliaci.github.io/NanosoldierReports/pkgeval_badges/report.html)
```
Comment (nickrobinson251) [inline README.md]: I was wrong about these. they suck
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -1,15 +1,16 @@
...
+[![code style blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
```
Comment (nickrobinson251) [inline README.md:9]: @oxinabox likes badges that don't change colour to be on a different line to those that do
```suggestion

[![code style blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
```

**Review by nickrobinson251 — CHANGES_REQUESTED** [review]
needs `@ref`s

File: docs/src/testutils.md (hunk 1 above)
Comment (glennmoy) [reply docs/src/testutils.md]: Ah I forgot that they work inside the docs just not the README. My bad.

**Comment by nickrobinson251** [comment]
Need to update `docs/make.jl`

from `repo="https://gitlab.invenia.ca/invenia/Models.jl/blob/{commit}{path}#L{line}"` to `repo=repo="https://github.com/invenia/Models.jl/blob/{commit}{path}#L{line}",`

**Comment by nickrobinson251** [comment]
there's a `(ref)` missing the `@` on [L43 of src/Models.jl](https://github.com/invenia/Models.jl/blob/gm/readme/src/Models.jl#L43)

File: docs/src/design.md [hunk 9]
```diff
@@ -1,8 +1,6 @@
 ## Design Documentation
```
Comment (nickrobinson251) [inline docs/src/design.md]: Should be title not level 2 header
```suggestion
# Design
```

File: docs/src/design.md [hunk 10]
```diff
@@ -13,22 +11,22 @@ BaselineModels exists to solve the issue highlighted by following quote:
 
 -- [Sculley et al 2015](https://papers.nips.cc/paper/5656-hidden-technical-debt-in-machine-learning-systems)
 
-BaselineModels provides a common API for mostly preexisting models to allow them to all be used in the same way.
+[Models.jl](https://github.com/invenia/Models.jl) provides a common API for mostly preexisting models to allow them to all be used in the same way.
 As such, the most important thing is that it itself has a common API.
 Here are some facts about that API:
 
 ### Models and Templates
```
Comment (nickrobinson251) [inline docs/src/design.md]: Need to be level 2 headers to appear in the sidebar
```suggestion
## Models and Templates
```

File: docs/src/design.md [hunk 11]
```diff
@@ -13,22 +11,22 @@ BaselineModels exists to solve the issue highlighted by following quote:
...
-The template based API is superior to these as it means `fit` is always the same.
-One does not have to carry both a model type, and a varying collection of keyword arguments, which would get complicated when composing wrapper models.
+All information about how to perform [`fit`](@ref), such as hyper-parameters, is stored inside the [`Template`](@ref).
+This is different from some other APIs which might, for example, pass hyper-parameters as keyword arguments to [`fit`](@ref).
+The [`Template`](@ref) based API is superior to these as it means [`fit`](@ref) is always the same.
+One does not have to carry both a [`Model`](@ref) type, and a varying collection of keyword arguments, which would get complicated when composing wrapper models.
 
 
-### `fit` and `predict`
+### Calling fit and predict
```
Comment (nickrobinson251) [inline docs/src/design.md]:
```suggestion
## Calling `fit` and `predict`
```

File: docs/src/design.md [hunk 12]
```diff
@@ -46,30 +44,30 @@ outputs = StatsBase.predict(
...
- - The functions must accept any `AbstractMatrix` for the `inputs` and `outputs` (`fit` only). If the underlying implementation needs a plain dense `Matrix` then `fit`/`predict` should perform the conversion.
- - `fit` always accepts a `weights` argument. If the underlying model does not support weighted fitting, then `fit` should throw and error if the weights that passed in and are not all equal.
- - `fit`/`predict` take no keyword arguments, or any other arguments except the ones shown.
+ - The functions must accept any `AbstractMatrix` for the `inputs` and `outputs` ([`fit`](@ref) only). If the underlying implementation needs a plain dense `Matrix` then [`fit`](@ref)/[`predict`](@ref) should perform the conversion.
+ - [`fit`](@ref) always accepts a `weights` argument. If the underlying [`Model`](@ref) does not support weighted fitting, then [`fit`](@ref) should throw and error if the weights that passed in and are not all equal.
+ - [`fit`](@ref)/[`predict`](@ref) take no keyword arguments, or any other arguments except the ones shown.
 
 ### Traits
```
Comment (nickrobinson251) [inline docs/src/design.md]:
```suggestion
## Traits
```

**Comment by nickrobinson251** [comment]
Need to add [`deploydocs`](https://juliadocs.github.io/Documenter.jl/dev/lib/public/#Documenter.deploydocs) to bottom of `docs/make.jl`

```julia
deploydocs(;
    repo="github.com/invenia/Models.jl",
    push_preview=true,
)
```

And also follow the [instructions on hosting docs](https://juliadocs.github.io/Documenter.jl/dev/man/hosting/#Hosting-Documentation-1)

**Comment by glennmoy** [comment]
I

**Comment by glennmoy** [comment]
> Need to add [`deploydocs`](https://juliadocs.github.io/Documenter.jl/dev/lib/public/#Documenter.deploydocs) to bottom of `docs/make.jl`
> 
> ```julia
> deploydocs(;
>     repo="github.com/invenia/Models.jl",
>     push_preview=true,
> )
> ```
> 
> And also follow the [instructions on hosting docs](https://juliadocs.github.io/Documenter.jl/dev/man/hosting/#Hosting-Documentation-1)

Nothing to add to the YAML file from what I can see. I just need to add ssh deploy keys?

File: README.md [hunk 13]
```diff
@@ -1,15 +1,15 @@
 # Models
 
-[![Latest](https://img.shields.io/badge/docs-latest-blue.svg)](https://invenia.pages.invenia.ca/research/Models.jl/)
-[![Coverage](https://gitlab.invenia.ca/invenia/Models.jl/badges/master/coverage.svg)](https://gitlab.invenia.ca/invenia/Models.jl/commits/master)
+[![Build Status](https://travis-ci.com/invenia/Models.jl.svg?branch=master)](https://travis-ci.com/invenia/Models.jl)
```
Comment (nickrobinson251) [inline README.md:6]:
```suggestion
[![Stable](https://img.shields.io/badge/docs-latest-blue.svg)](https://invenia.github.io/Models.jl/stable/)
[![Latest](https://img.shields.io/badge/docs-latest-blue.svg)](https://invenia.github.io/Models.jl/dev/)

[![Build Status](https://travis-ci.com/invenia/Models.jl.svg?branch=master)](https://travis-ci.com/invenia/Models.jl)
```

**Review by nickrobinson251 — APPROVED** [review]

**Pull request merged**